### PR TITLE
Migrate logs repo from travisCI to GitHub actions

### PR DESCRIPTION
**Pull request by jotompki**
*Issue #, if available:*

*Description of changes:*  Migrates the travis continuous integration jobs to GitHub actions. Similar to: https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/pull/108.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/pr-ci.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      run: pip install cloudformation-cli-java-plugin
+    - name: install and run pre-commit
+      uses: pre-commit/action@v2.0.0
+      with:
+        extra_args: --all-files
+    - name: Run maven verify for all resources
+      run: |
+        for directory in ./aws-*; do
+          cd "$directory"
+          mvn -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B clean verify || travis_terminate 1
```
Comment (jotompki) [inline .github/workflows/pr-ci.yml:38]:
```suggestion
          mvn -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B clean verify
```

**Review by sungkkim — APPROVED** [review]

**Review by omkhegde — APPROVED** [review]

**Review by PatMyron — CHANGES_REQUESTED** [review]
hundreds of [`aws-cloudformation-resource-providers`](https://github.com/aws-cloudformation/?q=aws-cloudformation-resource-providers-) repos to migrate and we've already gone through a lot of CI PR churn for them (all of @miparnisari's PRs to make sure Maven builds run instead of just `pre-commit` formatting, switching between CodeBuild, Travis, Github Actions, etc), so let's migrate all of them together

I'd prefer if we took the chance to improve these before starting to manually migrate hundreds of repos one-by-one. These should also be running [`cfn validate` and `cfn submit --dry-run` at the very least and arguably `cfn test`](https://github.com/aws-cloudformation/cloudformation-cli) and [`git-secrets`](https://github.com/awslabs/git-secrets) as well

**Comment by jotompki** [comment]
> hundreds of [`aws-cloudformation-resource-providers`](https://github.com/aws-cloudformation/?q=aws-cloudformation-resource-providers-) repos to migrate and we've already gone through a lot of CI PR churn for them (all of @miparnisari's PRs to make sure Maven builds run instead of just `pre-commit` formatting, switching between CodeBuild, Travis, Github Actions, etc), so let's migrate all of them together
> 
> I'd prefer if we took the chance to improve these before starting to manually migrate hundreds of repos one-by-one. These should also be running [`cfn validate` and `cfn submit --dry-run` at the very least and arguably `cfn test`](https://github.com/aws-cloudformation/cloudformation-cli) and [`git-secrets`](https://github.com/awslabs/git-secrets) as well

Doing a quick spot check it appears only 12/38 public resource provider repositories have a travis build job. All for automation of this process but we would probably have to schedule some dedicated work to do a big overhaul. 

The process of team commits to repo -> travisCI build doesn't work -> looks to documentation as to why -> sees example github action -> upgrades doesn't seem too bad for now.

**Comment by PatMyron** [comment]
> Doing a quick spot check it appears only 12/38 public resource provider repositories have a travis build job

Only a dozen repos even meeting our best practices from 2019 feels like even more reason to centralize the process longterm

**Review by miparnisari — APPROVED** [review]

**Comment by markkuhn** [comment]
This PR has been open for more than a year. Can we close it?

**Pull request closed without merging**